### ISSUE TITLE
Mixxx 1.8 beta1 hotcue saving error and crash when using Track editor

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#329]
Step 1
Load mp3, set one ore more Hotcues to the track. 

Step 2
Close and reopen Mixxx, select same track and open Trackeditor via right-click.

Actual Result
*Only one Hotcue (even when more are set) is shown with the wrong position
*When editing or adding Hotcue via Trackeditor Mixxx`crash.

Expected Result
* All Hotcues to be shown with the correct position
* All Hotcues editable without hang and crash

See screenshot & crashlog

**Comment by Holzhaus** [comment]
(by jus)

**Comment by Holzhaus** [comment]
(by jus)

**Comment by Holzhaus** [comment]
(by rryan)
This should be fixed now. Verified fine for me on the Deere skin -- hotcues in the overview line up with their actual position now and the track information dialog shows them properly and doesn't crash when you click apply.

**Comment by Holzhaus** [comment]
(by jus)
Nice,the crash is gone :-) But there may some minor errors left.

Track information dialog:
*Position for (Hot)cues is shown twice the actual value. A (Hot)cue set at 0:10.00 shows up as 0:20.00.
*You can not delete or activate (Hot)Cues via the the corresponding buttons in track information dialog.
*If a (Hot)cue is deleted via GUI, the position remains in the database. Only the (Hot)cue number is set to -1.

see screenshot

**Comment by Holzhaus** [comment]
(by rryan)
Hey Jus,

I just fixed the doubled-position issue and the delete button. I don't think that activate will work for this release because it's ambiguous which deck the person means for it to play on. (the same track could be loaded in both decks). I'll probably remove it. 

Not sure if you noticed, but in trunk no changes to tracks or cues save. I revived the features-sqlite branch to fix this problem mixxx-wide. 

When you delete a hotcue via hotcue_xxx_clear, it just resets the cue's hotcue to -1. This is intentional so that if you spent time setting up your tracks hotcue's (adding labels, etc), then accidentally right-clicking a hotcue button will never actually delete it. The track info dialog shouldn't show the -1 though, it should just show it as blank.

Thanks,
RJ
